### PR TITLE
Refactor QueryFeatureActivity to use SymbolLayer instead of MarkerViewOptions

### PR DESCRIPTION
**Pull request by langsmith**
Resolves https://github.com/mapbox/mapbox-android-demo/issues/1144 by refactoring `QueryFeatureActivity` to use `SymbolLayer` instead of `MarkerViewOptions`.

ezgif com-resize (11)

### CODE REVIEW

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/query/QueryFeatureActivity.java [hunk 1]
```diff
@@ -53,53 +78,233 @@ public void onMapReady(@NonNull final MapboxMap mapboxMap) {
...
+   * Call be optionally be called to update the underlying data source after execution.
+   * </p>
+   * <p>
+   * Generating Views on background thread since we are not going to be adding them to the view hierarchy.
+   * </p>
+   */
+  private static class GenerateViewIconTask extends AsyncTask<FeatureCollection, Void, HashMap<String, Bitmap>> {
 
-      String property;
+    private final HashMap<String, View> viewMap = new HashMap<>();
```
Comment (LukasPaczos) [inline MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/query/QueryFeatureActivity.java]: `viewMap` can be removed.

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/query/QueryFeatureActivity.java [hunk 2]
```diff
@@ -53,53 +78,233 @@ public void onMapReady(@NonNull final MapboxMap mapboxMap) {
...
-    mapboxMap.selectMarker(featureMarker);
 
-    return true;
+    @Override
+    protected void onPostExecute(HashMap<String, Bitmap> bitmapHashMap) {
+      super.onPostExecute(bitmapHashMap);
+      QueryFeatureActivity activity = activityRef.get();
+      if (activity != null && bitmapHashMap != null) {
+        activity.setImageGenResults(bitmapHashMap);
+        activity.refreshSource(featureAtMapClickPoint);
```
Comment (LukasPaczos) [inline MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/query/QueryFeatureActivity.java]: The source doesn't need to be refreshed at this point. Adding an image will update the map if necessary.

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/query/QueryFeatureActivity.java (hunk 1 above)
Comment (langsmith) [reply MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/query/QueryFeatureActivity.java]: Removed

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/query/QueryFeatureActivity.java (hunk 2 above)
Comment (langsmith) [reply MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/query/QueryFeatureActivity.java]: Seems that it does need to be refreshed. When I remove `activity.refreshSource(featureAtMapClickPoint);`  and run the example, the property list info window icon doesn't appear when the map is tapped on.

Comment (LukasPaczos) [reply MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/query/QueryFeatureActivity.java]: Ah, I thought it's also updated during the click itself.

**Pull request merged**